collections/src/main/scala/strawman/collection/immutable/SortedMap.scala

```diff
@@ -40,15 +61,55 @@ trait SortedMapOps[K, +V, +CC[X, +Y] <: Map[X, Y] with SortedMapOps[X, Y, CC, _]
 
     // We override these methods to fix their return type (which would be `Map` otherwise)
     def updated[V1 >: V](key: K, value: V1): CC[K, V1]
-    @`inline` final override def + [V1 >: V](kv: (K, V1)): CC[K, V1] = updated(kv._1, kv._2)
+    @`inline` final override def +[V1 >: V](kv: (K, V1)): CC[K, V1] = updated(kv._1, kv._2)
 
     override def concat[V2 >: V](xs: collection.Iterable[(K, V2)]): CC[K, V2] = {
         var result: CC[K, V2] = coll
         val it = xs.iterator()
         while (it.hasNext) result = result + it.next()
         result
     }
-
 }
 
-object SortedMap extends SortedMapFactory.Delegate[SortedMap](TreeMap)
\ No newline at end of file
+object SortedMap extends SortedMapFactory.Delegate[SortedMap](TreeMap) {
+
+  final class WithDefault[K, +V](underlying: SortedMap[K, V], defaultValue: K => V)
+    extends Map.WithDefault[K, V](underlying, defaultValue)
+    with SortedMap[K, V]
+    with SortedMapOps[K, V, SortedMap, WithDefault[K, V]]{
+
+    implicit def ordering: Ordering[K] = underlying.ordering
+
+    def sortedMapFactory: SortedMapFactory[SortedMap] = underlying.sortedMapFactory
+
+    def iteratorFrom(start: K): strawman.collection.Iterator[(K, V)] = underlying.iteratorFrom(start)
+
+    protected[this] def sortedMapFromIterable[K2, V2](it: strawman.collection.Iterable[(K2, V2)])(implicit ordering: Ordering[K2]): SortedMap[K2, V2] =
+      sortedMapFactory.from(it)
+
+    def keysIteratorFrom(start: K): strawman.collection.Iterator[K] = underlying.keysIteratorFrom(start)
+
+    def rangeImpl(from: Option[K], until: Option[K]): WithDefault[K, V] =
+      new WithDefault[K, V](underlying.rangeImpl(from, until), defaultValue)
+
+    // Need to override following methods to match type signatures of `SortedMap.WithDefault`
+    // for operations preserving default value
+
+    override protected[this] def mapFromIterable[K2, V2](it: collection.Iterable[(K2, V2)]): Map[K2, V2] =
+      mapFactory.from(it)
```

Comment:
It looks like this is the same implementation as in the super type.

Reply:
Yeah, actually I had to add it purposefully with `override`. `SortedMap.WithDefault` is extending `SortedMap' and `Map.WithDefault`. But since `SortedMap` already implements `mapFromIterable`, there is collision happening in the inherited members of `SortedMap.WithDefault`.  See [this in implementation of `mapFromIterable` in `SortedMap`](https://github.com/scala/collection-strawman/blob/1660bb51be0175bd6b64758727e95b5d8c096d30/collections/src/main/scala/strawman/collection/immutable/SortedMap.scala#L38)

Reply:
also, super type always returns `Map.from(it)`. Whereas, default implementations both in mutable and immutable return map created from `underlying.mapFactory.From(it)`. This makes sure that, whenever `mapFromIterable` is called on withDefault type, it will return map from underlying map's factory than plain `Map.from(it)`.  Based on [this feedback](https://github.com/scala/collection-strawman/pull/381/files/83fa419316f90b1d7f11003cb0000a0ae146da57#diff-d2ae80cbea21c68573a69ce27e2d0323R152) I thought, it's better to `override` it in WithDefault implementation than making super type's implementation final.